discovery-aws-api/src/main/scala/akka/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscovery.scala

```diff
@@ -40,21 +41,22 @@ import scala.concurrent.{ ExecutionContext, Future }
 
 }
 
-class Ec2TagBasedSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
+class Ec2TagBasedSimpleServiceDiscovery(system: ExtendedActorSystem) extends SimpleServiceDiscovery {
 
   private val log = Logging(system, classOf[Ec2TagBasedSimpleServiceDiscovery])
 
-  private val ec2Client: AmazonEC2 = {
-    val clientConfiguration = new ClientConfiguration()
-    // we have our own retry/back-off mechanism, so we don't need EC2Client's in addition
-    clientConfiguration.setRetryPolicy(PredefinedRetryPolicies.NO_RETRY_POLICY)
-    AmazonEC2ClientBuilder.standard().withClientConfiguration(clientConfiguration).build()
-  }
-
   private implicit val ec: ExecutionContext = system.dispatchers.lookup("akka.actor.default-blocking-io-dispatcher")
 
   private val config = system.settings.config.getConfig("akka.discovery.aws-api-ec2-tag-based")
 
+  private val clientConfigFqcn: Option[String] = { // FQCN of a class that extends com.amazonaws.ClientConfiguration
+    if (config.hasPath("client-config")) {
```

Comment:
I would prefer to have this in `reference.conf`:
```
akka.discovery.aws-api-ec2-tag-based.client-config = ""
```

with comment explaining what it is and that it can be defined to a fqcn of a class implementing `ClientConfiguration` with a default constructor (no parameters).

In that way `reference.conf` becomes the complete source/documentation of what configuration is possible.

By the way, should it support an `ActorSystem` constructor parameter? Perhaps it might need to read additional config via `system.settings.config` or use logging or something?